lib/engine/step/g_18_al/assign.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require_relative '../assign'
+
+module Engine
+  module Step
+    module G18AL
+      class Assign < Assign
+        def process_assign(action)
+          company = action.entity
+          raise GameError, "#{company.owner.name} owns no trains" if company.owner.trains.empty?
+
+          target = action.target
+          hexes = company.abilities(:assign_hexes)&.hexes
+          reachable = @game.graph.reachable_hexes(company.owner).find { |h, _| h.id == target.id }
+          raise GameError, "#{@game.get_location_name(target.id)} is not reachable" unless reachable
+
+          super
+
+          # Add a revenue bonus for the corporation
+          # that will be removed in phase 6
+          ability = Engine::Ability::HexBonus.new(
+            type: :hexes_bonus,
+            description: "Coal Field token: #{@game.get_location_name(target.id)}",
+            hexes: [target.id],
+            amount: 10
+          )
+          company.owner.add_ability(ability)
+          @game.hexes.select { |hex| hexes.include?(hex.name) }
+            .each { |hex| hex.tile&.icons = [] }
```

Comment:
doesn't a hex always have a tile? have you seen this break?

Reply:
I have now removed the ampersand. Will change if it breaks.